synapse/lib/stormtypes.py

```diff
@@ -7294,43 +7306,54 @@ async def _methTriggerAdd(self, tdef):
 
     async def _methTriggerDel(self, prefix):
         useriden = self.runt.user.iden
-        viewiden = self.runt.snap.view.iden
         trig = await self._matchIdens(prefix)
         iden = trig.iden
 
         todo = s_common.todo('delTrigger', iden)
         gatekeys = ((useriden, ('trigger', 'del'), iden),)
-        await self.dyncall(viewiden, todo, gatekeys=gatekeys)
+        await self.dyncall(trig.view.iden, todo, gatekeys=gatekeys)
 
         return iden
 
     async def _methTriggerMod(self, prefix, query):
         useriden = self.runt.user.iden
-        viewiden = self.runt.snap.view.iden
         query = await tostr(query)
         trig = await self._matchIdens(prefix)
         iden = trig.iden
         gatekeys = ((useriden, ('trigger', 'set'), iden),)
         todo = s_common.todo('setTriggerInfo', iden, 'storm', query)
-        await self.dyncall(viewiden, todo, gatekeys=gatekeys)
+        await self.dyncall(trig.view.iden, todo, gatekeys=gatekeys)
 
         return iden
 
-    async def _methTriggerList(self):
-        view = self.runt.snap.view
-        triggers = []
+    async def _methTriggerList(self, all=False):
+        if all:
+            views = self.runt.snap.core.listViews()
+        else:
+            views = [self.runt.snap.view]
 
-        for iden, trig in await view.listTriggers():
-            if not allowed(('trigger', 'get'), gateiden=iden):
+        triggers = []
+        for view in views:
+            if not allowed(('view', 'read'), gateiden=view.iden):
                 continue
-            triggers.append(Trigger(self.runt, trig.pack()))
+
+            for iden, trig in await view.listTriggers():
+                if not allowed(('trigger', 'get'), gateiden=iden):
+                    continue
+                triggers.append(Trigger(self.runt, trig.pack()))
 
         return triggers
 
     async def _methTriggerGet(self, iden):
-        trigger = await self.runt.snap.view.getTrigger(iden)
+        trigger = None
+        for view in self.runt.snap.core.listViews():
```

Comment:
I wonder if there is a fast-path here of "check my current view triggers first" before spinning through the list of views to get a trigger?